src/ui/pages/common/PageForm.js

```diff
@@ -239,21 +239,34 @@ export class PageFormBody extends Component {
         <Row>
           <Col xs={12}>
             <div className="btn-toolbar pull-right">
+
               <Button
                 className="PageForm__save-and-configure-btn"
                 type="submit"
                 bsStyle="success"
                 disabled={invalid || submitting}
+                onClick={handleSubmit(values =>
+                  this.props.onSubmit({
+                    ...values,
+                    action: ACTION_SAVE_AND_CONFIGURE,
+                  }))}
               >
                 <FormattedMessage id="pages.pageForm.saveAndConfigure" />
+
               </Button>
               <Button
                 className="PageForm__save-btn"
                 type="submit"
                 bsStyle="primary"
                 disabled={invalid || submitting}
+                onClick={handleSubmit(values =>
+                  this.props.onSubmit({
+                    ...values,
+                    action: ACTION_SAVE,
```

Comment:
These values are coming from `redux-form`, we can't be sure they won't be handled internally (e.g. for validation) therefore potentially causing unexpected behaviours. You should add a second parameter to onSubmit, instead of modifying the values object.